app/controllers/submissions_controller.rb

```diff
@@ -72,6 +73,11 @@ def download_file
     end
   end
 
+  def finalize
+    @submission.update(cause: 'submit')
```

Comment:
Since we have no other guard clause here, I would use the bang method:

```suggestion
    @submission.update!(cause: 'submit')
```